src/domain/accounts/domain/src/use_cases/update_account_use_case.rs

```diff
@@ -10,37 +10,44 @@
 use internal_error::InternalError;
 use thiserror::Error;
 
-use crate::{Account, AccountErrorDuplicate};
+use crate::{Account, AccountErrorDuplicate, AccountNotFoundByIdError, UpdateAccountError};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #[async_trait::async_trait]
-pub trait RenameAccountUseCase: Send + Sync {
-    async fn execute(
-        &self,
-        account: &Account,
-        new_name: odf::AccountName,
-    ) -> Result<(), RenameAccountError>;
+pub trait UpdateAccountUseCase: Send + Sync {
+    async fn execute(&self, account: &Account) -> Result<(), UpdateAccountUseCaseError>;
 }
 
-////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-// Errors
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #[derive(Debug, Error)]
-pub enum RenameAccountError {
-    #[error(transparent)]
-    Duplicate(AccountErrorDuplicate),
-
+pub enum UpdateAccountUseCaseError {
     #[error(transparent)]
     Access(
         #[from]
         #[backtrace]
         odf::AccessError,
     ),
 
+    #[error(transparent)]
+    NotFound(AccountNotFoundByIdError),
+
+    #[error(transparent)]
+    Duplicate(AccountErrorDuplicate),
+
     #[error(transparent)]
     Internal(#[from] InternalError),
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+impl From<UpdateAccountError> for UpdateAccountUseCaseError {
+    fn from(err: UpdateAccountError) -> Self {
+        match err {
+            UpdateAccountError::NotFound(e) => UpdateAccountUseCaseError::NotFound(e),
+            UpdateAccountError::Duplicate(e) => UpdateAccountUseCaseError::Duplicate(e),
+            UpdateAccountError::Internal(e) => UpdateAccountUseCaseError::Internal(e),
+        }
+    }
+}
```

Comment:
If that's because of "Access" only, it's easier to have just 1 error type both for use case and service.